synth_node_bin/src/main.rs

```diff
@@ -0,0 +1,128 @@
+//! A synthetic node binary can be used to interact with the node in the
+//! background from a different runtime environment.
+//!
+//! This is a quick solution - in future Ziggurat projects it will be done differently.
+//!
+//! Remaining worklist (probably not going to be implemented in this repo):
+//!   - Add an argument option to choose specific action in order to support
+//!     different synthetic node binary implementations at once. A few examples:
+//!     ```
+//!        ./synthetic_node_bin --action=A    // Runs an idle/friendly synthetic node
+//!        ./synthetic_node_bin --action=B    // Runs a wild synthetic node which does something funny
+//!     ```
+use std::{net::SocketAddr, process::ExitCode};
+use tokio::time::{sleep, Duration};
+
+use clap::Parser;
+use ziggurat_xrpl::tools::{config::SynthNodeCfg, synth_node::SyntheticNode};
+use anyhow::Result;
+
+/// A synthetic node which can connect to the XRPL node and preform some
+/// actions independently.
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct CmdArgs {
+    /// An address of the node in the <ip>:<port> format.
+    #[arg(short = 'i', long)]
+    node_ip: Option<SocketAddr>,
```

Comment:
Maybe `node_addr` as it's not `IpAddr` type but `SocketAddr` so with port.

Reply:
Done